dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/InclusionBasedPointerAnalysis.java

```diff
@@ -0,0 +1,967 @@
+                    }
+                }
+                processCommunication(variableStores, loads);
+                processCommunication(stores, variableLoads);
+            }
+        }
+    }
+
+    // Removes information from the internal graph, which are no longer needed after the algorithm has finished.
+    // This simplifies alias queries and releases memory resources.
+    private void postProcess(Map.Entry<MemoryCoreEvent, DerivedVariable> entry) {
+        logger.trace("{}", entry);
+        final DerivedVariable address = entry.getValue();
+        if (address == null) {
+            // should have already warned about this event
+            return;
+        }
+        // Remove all obsolete inclusion relationships between register states.
+        address.base.includes.removeIf(i -> i.source.object == null);
+        address.base.loads.clear();
+        address.base.stores.clear();
+        address.base.seeAlso.clear();
+        // In a well-structured program, all address expressions refer to at least one memory object.
+        if (logger.isWarnEnabled() && address.base.object == null &&
+                address.base.includes.stream().allMatch(i -> i.source.object == null)) {
+            logger.warn("empty pointer set for {}", synContext.get().getContextInfo(entry.getKey()));
+        }
+        if (address.base.includes.size() != 1) {
+            return;
+        }
+        final IncludeEdge includeEdge = address.base.includes.get(0);
+        final Modifier modifier = join(includeEdge.modifier, address.modifier);
+        assert includeEdge.source.object != null;
+        // If the only included address refers to the last element, treat it as a direct static offset instead.
+        final int remainingSize = includeEdge.source.object.size() - modifier.offset;
+        for (final Integer a : modifier.alignment) {
+            if (a < remainingSize) {
+                return;
+            }
+        }
+        entry.setValue(derive(includeEdge.source, modifier.offset, List.of()));
+    }
+
+    // ================================ Internals ================================
+
+    private static final class Variable {
+        // Any value contained in these Variables is also contained (+ offset).
+        private final List<IncludeEdge> includes = new ArrayList<>();
+        // There is some load event using this (+ offset) as pointer set and the other variable as output set.
+        private final List<LoadEdge> loads = new ArrayList<>();
+        // There is some store event using this (+ offset1) as pointer set and the other variable (+ offset2) as value set.
+        private final List<StoreEdge> stores = new ArrayList<>();
+        // All variables that have a direct (includes/loads/stores) link to this.
+        private final Set<Variable> seeAlso = new HashSet<>();
+        // If nonnull, this variable represents that object's base address.
+        private final MemoryObject object;
```

Comment:
Most comments here are outdated. I also think that you can put most of the descriptions on the corresponding record classes (i.e., the explanation of the meaning of the edges).
I would just make sure that these comments (or even the variable names) make clear in which direction the edges go: ingoing or outgoing.